pkg/controllers/osc/osc_controller.go

```diff
@@ -86,29 +92,31 @@ func Add(
 	podCIDR string,
 	nodePortRange string) error {
 	reconciler := &Reconciler{
-		Client:                mgr.GetClient(),
-		log:                   log,
-		namespace:             namespace,
-		clusterAddress:        clusterName,
-		generator:             generator,
-		kubeconfig:            kubeconfig,
-		clusterDNSIPs:         clusterDNSIPs,
-		containerRuntime:      containerRuntime,
-		pauseImage:            pauseImage,
-		initialTaints:         initialTaints,
-		externalCloudProvider: externalCloudProvider,
-		nodeHTTPProxy:         nodeHTTPProxy,
-		nodeNoProxy:           nodeNoProxy,
-		podCIDR:               podCIDR,
-		nodePortRange:         nodePortRange,
+		log:                     log,
+		workerClient:            mgr.GetClient(),
```

Comment:
I guess this is not quite correct. AFAIK, the client is configured via the passed config to the manager by called the `New()` function. Thus, this would give you a client for seed not for the workers clusters.

Reply:
OSC controller requires manager for `worker` and I'm passing the `workerManager` in this case so this should work. 

```go
	// Setup OSC controller
	if err := osc.Add(
		workerMgr,
		log,
```

But for clarity, I can rename `mgr` to `workerMgr`.

Reply:
Not really, you are passing the original manager:
https://github.com/kubermatic/operating-system-manager/pull/106/files#diff-9e1ff01bcff629057975a3156f469ee57167759cb83ff542205cb41c134b4b7eR191

Reply:
Ah ok my bad!

Reply:
You are confusing 2 different values I think. This is how the eventual assignments would look like for the OSC controller:

```
mgr = workerManager (WORKER)
workerClient = workerManager.GetClient() (WORKER)
client = client (SEED)
```

https://github.com/kubermatic/operating-system-manager/pull/106/files#diff-9e1ff01bcff629057975a3156f469ee57167759cb83ff542205cb41c134b4b7eR189

Reply:
No problem, at least I got to revisit and verify everything again :D